client/src/components/RecipeList/components/RecipeCard.jsx

```diff
@@ -1,42 +1,195 @@
-          src={imageURL}
-          alt={title}
-        />
-      )}
-    >
-      <Meta
-        className="recipe-card-meta"
-        title={title}
-      />
-    </Card>
-  );
-};
+
+  showModal() {
+    this.setState({
+      showModal: true,
+    });
+  }
+
+  hideModal() {
+    this.setState({
+      showModal: false,
+    });
+  }
+
+
+  render() {
+    const { recipe } = this.props;
+    const {
+      title, link, image: imageURL, total_time: totalTime, rating, number_of_raters: numberOfRaters,
+      tags, id,
+    } = recipe;
+    const viewMoreText = `${tags.length - 5} more`;
+    const { showModal, viewMore } = this.state;
+
+    const openRecipe = () => {
+      window.open(link, 'noopener noreferrer');
+    };
+
+    const findSite = () => Object.entries(RecipeNames)
+      .find((pair) => new RegExp(pair[0]).test(id))[1];
+
+    const site = findSite();
+
+    return (
+      <>
+        <Card
+          className="recipe-card"
+          // onClick={() => openRecipe(link)}
+          hoverable
+          cover={(
+            <img
+              className="recipe-image"
+              src={imageURL}
+              alt={title}
+            />
+          )}
+
+          actions={[
+            <Button className="recipe-button" block type="link" size="large" onClick={this.showModal}>
+              <InfoCircleOutlined />
+            </Button>,
+            <Button className="recipe-button" block type="link" size="large">
+              <StarOutlined />
+            </Button>,
+            <Button className="recipe-button" block type="link" size="large" onClick={openRecipe}>
+              <ExportOutlined />
+            </Button>,
+          ]}
+        >
+          <Meta
+            className="recipe-card-meta"
+            title={title}
+            description={site}
+          />
+        </Card>
+        <Modal
+          title={title}
+          visible={showModal}
+          onCancel={this.hideModal}
+          footer={[
+            <Button key="Save">
+              <StarOutlined />
+            </Button>,
+            <Button key="Go" onClick={openRecipe}>
+              <ExportOutlined />
+            </Button>,
+          ]}
+        >
+          <img
+            className="recipe-modal-image"
+            src={imageURL}
+            alt={title}
+          />
+
+          <Divider />
+          <Descriptions title="Recipe Info" layout="vertical" colon={false}>
+            <Descriptions.Item label="Website">{ site }</Descriptions.Item>
+            <Descriptions.Item label="Cooking Time">
+              {totalTime ? `${totalTime} minutes` : (
+                <span>
+                  {' '}
+                  <CloseCircleOutlined />
+                  {' '}
+                  Unavailable
+                  {' '}
+                </span>
+              ) }
+            </Descriptions.Item>
+
+            <Descriptions.Item label="Rating">
+              { rating ? (
+                <span>
```

Comment:
I would rather that you replace this with the ant grid if you can. It'll look nicer. You need a row with three equal columns which is easy to configure with that component.